client/packages/common/src/ui/layout/tables/columns/types.ts

```diff
@@ -90,6 +90,9 @@ export interface Column<T extends RecordWithId> {
   backgroundColor?: string;
 
   Cell: JSXElementConstructor<CellProps<T>>;
+  // For passing additional props to the above Cell
+  cellProps?: Record<string, unknown>;
```

Comment:
Hm ... I played around with something like this: 

` cellProps?: Omit<ComponentProps<Column<T>['Cell']>, keyof CellProps<T>>;`

It's not quite right but might be good to experiment with for a sec? Would be nice for cellProps to be typed based on the Cell

Reply:
Yeah, I had it *almost* working too. I'll have another look, won't spend too long on it.

Reply:
thanks - that was my concern too. allowing untyped spreading and working around the type system isn't ideal

Reply:
The difficulty is that the `Cell` prop itself isn't actually correctly defined: `JSXElementConstructor<CellProps<T>>` but it can can have props beyond `CellProps<T>`, so it's hard to define props type for props the `Cell` property isn't actually aware of (if that makes any sense). 

I tried passing in a second generic `U` to the Column definition, but this requires every column to have the same props for its "Cell" which is obviously not the case. 

I think we inherently lose type information by defining `Cell` like this. Any other suggestions?

Reply:
For now I've just added a comment to use with caution. I think a more significant refactor is in order to properly type these 🤔